specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json

```diff
@@ -365,6 +372,100 @@
           "maximum": 86400
         }
       }
+    },
+    "TriggerReference": {
+      "description": "Trigger reference type.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Trigger reference type.",
+          "type": "string",
+          "enum": [
+            "TriggerReference"
+          ]
+        },
+        "referenceName": {
+          "description": "Reference trigger name.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type",
+        "referenceName"
+      ]
+    },
+    "DependencyReference": {
+      "description": "Referenced dependency",
+      "type": "object",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "description": "The type of dependency reference",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "TriggerDependencyReference": {
+      "description": "Trigger referenced dependency",
+      "type": "object",
+     "allOf": [
+        {
+          "$ref": "#/definitions/DependencyReference"
+        }
+      ],
+      "properties": {
+        "referenceTrigger": {
+          "description": "Referenced trigger",
+          "$ref": "#/definitions/TriggerReference"
+        }
+      },
+      "required": [
+        "referenceTrigger"
+      ]
+    },
+    "TumblingWindowTriggerDependencyReference": {
+      "description": "Referenced tumbling window trigger dependency",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TriggerDependencyReference"
+        }
+      ],
+      "properties": {
+        "offset": {
+          "description": "Timespan applied to the start time of a tumbling window when evaluating dependency, pattern: ((\\d+)\\.)?(\\d\\d):(60|([0-5][0-9])):(60|([0-5][0-9])).",

```

Comment:
>((\\d+)\\.)?(\\d\\d):(60|([0-5][0-9])):(60|([0-5][0-9])) [](start = 119, length = 56)

I think you can add this as validation, then SDK will have a code for that. Check the parameters section in [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json)